### PR TITLE
Expression editor improvements

### LINKED ISSUE #713: ExpressionEditor improvements for NodeDefEditView
- [x] In Identifier variables dropdown first variable must be nodeDef current ( see #703 )
- [x] In Identifier variables dropdown use variable names, not labels ( see #701 )

### PR DESCRIPTION
**Pull request by SteRiccio**
Resolves #713

### CODE REVIEW

File: webapp/commonComponents/expression/expressionEditorPopupState.js [hunk 1]
```diff
@@ -121,14 +121,14 @@ export const useAdvancedExpressionEditorPopupState = props => {
 
 export const mapStateToProps = (state, props) => {
```
Comment (minotogna) [inline webapp/commonComponents/expression/expressionEditorPopupState.js:122]: is it possible to get rid of mapStateToProps?

File: webapp/commonComponents/expression/expressionVariables.js [hunk 2]
```diff
@@ -8,37 +8,50 @@ import { sqlTypes } from '@common/surveyRdb/sqlTypes'
...
+  if (NodeDef.isDecimal(nodeDef)) {
+    return sqlTypes.decimal
+  }
+  return sqlTypes.varchar
+}
+
+// Returns the name of the given nodeDef and appends "THIS" if it's equal to the current one
+const getVariableLabel = (nodeDef, nodeDefCurrent, i18n) =>
+  NodeDef.getName(nodeDef) +
+  (nodeDefCurrent && NodeDef.isEqual(nodeDef)(nodeDefCurrent) ? ` (${i18n.t('expressionEditor.this')})` : '')
```
Comment (minotogna) [inline webapp/commonComponents/expression/expressionVariables.js:24]: don't append THIS

File: webapp/commonComponents/expression/expressionVariables.js [hunk 3]
```diff
@@ -8,37 +8,50 @@ import { sqlTypes } from '@common/surveyRdb/sqlTypes'
...
-    // TODO: add uuid here for symmetry?
+    uuid: NodeDef.getUuid(nodeDef),
   },
 ]
 
-const getSqlVariables = (nodeDef, lang) => {
+const getSqlVariables = (nodeDef, nodeDefCurrent, i18n) => {
   const colNames = NodeDefTable.getColNames(nodeDef)
 
   // TODO: Explain what getLabel does and why
```
Comment (minotogna) [inline webapp/commonComponents/expression/expressionVariables.js:38]: please, remove this old TODO. not sure who added it and why it's there

File: webapp/commonComponents/expression/expressionVariables.js [hunk 4]
```diff
@@ -8,37 +8,50 @@ import { sqlTypes } from '@common/surveyRdb/sqlTypes'
...
-  const getLabel = col =>
-    NodeDef.getLabel(nodeDef, lang) + (colNames.length === 1 ? '' : ' - ' + NodeDefTable.extractColName(nodeDef, col))
+  const getLabel = (col) =>
+    getVariableLabel(nodeDef, nodeDefCurrent, i18n) +
+    (colNames.length === 1 ? '' : ` - ${NodeDefTable.extractColName(nodeDef, col)}`)
```
Comment (minotogna) [inline webapp/commonComponents/expression/expressionVariables.js:41]: The sql mode is used in Data vis. Not sure if there we should use labels or names? what do you think?

File: webapp/commonComponents/expression/expressionVariables.js [hunk 5]
```diff
@@ -52,30 +65,34 @@ const getChildDefVariables = (survey, nodeDefContext, nodeDefCurrent, mode, lang
...
-      const childVariables = getChildDefVariables(surveyWithDependencies, nodeDef, nodeDefCurrent, mode, lang)
+      const childVariables = getChildDefVariables(surveyWithDependencies, nodeDef, nodeDefCurrent, mode, i18n)
       variables.push(...childVariables)
     }
   })(surveyWithDependencies)
 
+  if (nodeDefCurrent) {
+    // Move current node def variable to the first position
+    const nodeDefCurrentUuid = NodeDef.getUuid(nodeDefCurrent)
+    variables.sort((varA, varB) => (varB.uuid === nodeDefCurrentUuid) - (varA.uuid === nodeDefCurrentUuid))
```
Comment (minotogna) [inline webapp/commonComponents/expression/expressionVariables.js:95]: besides the nodeDefCurrent to top, should we sort them alphabetically as well?

File: core/i18n/resources/en.js [hunk 6]
```diff
@@ -756,6 +756,7 @@ $t(common.cantUndoWarning)`,
     group: 'Group',
     var: 'Var',
     const: 'Const',
+    this: 'THIS',
```
Comment (minotogna) [inline core/i18n/resources/en.js:759]: remove this

File: webapp/commonComponents/expression/expressionVariables.js (hunk 2 above)
Comment (SteRiccio) [reply webapp/commonComponents/expression/expressionVariables.js:24]: it has been asked by Lauri... he wanted to show the variable related to the current node definition with something like `<THIS>`, what should we write instead?

File: webapp/commonComponents/expression/expressionVariables.js (hunk 4 above)
Comment (SteRiccio) [reply webapp/commonComponents/expression/expressionVariables.js:41]: I think we should be consistent and use the node definition names even there.

File: webapp/commonComponents/expression/expressionVariables.js (hunk 5 above)
Comment (SteRiccio) [reply webapp/commonComponents/expression/expressionVariables.js:95]: ok

File: webapp/commonComponents/expression/expressionVariables.js (hunk 2 above)
Comment (minotogna) [reply webapp/commonComponents/expression/expressionVariables.js:24]: I didn't see where Lauri asks to read 'This'.
Do not add anything

File: webapp/commonComponents/expression/expressionVariables.js (hunk 4 above)
Comment (minotogna) [reply webapp/commonComponents/expression/expressionVariables.js:41]: DataVis is for all users. Form designer is for some specific user.
DataVis also has sort where uses labels. 
I would still use labels in DataVis

File: webapp/commonComponents/expression/expressionVariables.js (hunk 2 above)
Comment (SteRiccio) [reply webapp/commonComponents/expression/expressionVariables.js:24]: ok

File: webapp/commonComponents/expression/expressionEditorPopupState.js (hunk 1 above)
Comment (SteRiccio) [reply webapp/commonComponents/expression/expressionEditorPopupState.js:122]: sure

File: webapp/commonComponents/expression/expressionVariables.js (hunk 3 above)
Comment (SteRiccio) [reply webapp/commonComponents/expression/expressionVariables.js:38]: ok

File: webapp/commonComponents/expression/expressionVariables.js (hunk 4 above)
Comment (SteRiccio) [reply webapp/commonComponents/expression/expressionVariables.js:41]: ok

File: core/i18n/resources/en.js (hunk 6 above)
Comment (SteRiccio) [reply core/i18n/resources/en.js]: ok

**Pull request merged**